dev/NumberBox/NumberBoxAutomationPeer.cpp

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "NumberBox.h"
+#include "NumberBoxAutomationPeer.h"
+#include "Utils.h"
+
+NumberBoxAutomationPeer::NumberBoxAutomationPeer(winrt::NumberBox const& owner)
+    : ReferenceTracker(owner)
+{
+}
+
+// IAutomationPeerOverrides
+winrt::IInspectable NumberBoxAutomationPeer::GetPatternCore(winrt::PatternInterface const& patternInterface)
+{
+    if (patternInterface == winrt::PatternInterface::RangeValue)
+    {
+        return *this;
+    }
+
+    return __super::GetPatternCore(patternInterface);
+}
+
+hstring NumberBoxAutomationPeer::GetClassNameCore()
+{
+    return winrt::hstring_name_of<winrt::NumberBox>();
+}
+
+winrt::AutomationControlType NumberBoxAutomationPeer::GetAutomationControlTypeCore()
+{
+    return winrt::AutomationControlType::Spinner;
```

Comment:
>Spinner [](start = 41, length = 7)

Is spinner the right type to return when we are in the SpinButtonsCollapsed state?

Reply:
I believe so, because it's the best match. To an accessibility user the control is the same whether or not the buttons are visible, so it makes sense to me.